src/go/k8s/controllers/redpanda/cluster_controller.go

```diff
@@ -129,62 +129,88 @@ func (r *ClusterReconciler) Reconcile(
 		}
 	}
 
+	err := r.reportStatus(ctx, &redpandaCluster, sts.LastObservedState, headlessSvc.HeadlessServiceFQDN(), nodeportSvc.Key())
+	if err != nil {
+		log.Error(err, "Unable to report status")
+	}
+
+	return ctrl.Result{}, err
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *ClusterReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&redpandav1alpha1.Cluster{}).
+		Owns(&appsv1.StatefulSet{}).
+		Owns(&corev1.Service{}).
+		Complete(r)
+}
+
+func (r *ClusterReconciler) reportStatus(
+	ctx context.Context,
+	redpandaCluster *redpandav1alpha1.Cluster,
+	lastObservedSts *appsv1.StatefulSet,
+	internalFQDN string,
+	nodeportSvcName types.NamespacedName,
+) error {
 	var observedPods corev1.PodList
 
 	err := r.List(ctx, &observedPods, &client.ListOptions{
-		LabelSelector: labels.ForCluster(&redpandaCluster).AsClientSelector(),
+		LabelSelector: labels.ForCluster(redpandaCluster).AsClientSelector(),
 		Namespace:     redpandaCluster.Namespace,
 	})
 	if err != nil {
-		log.Error(err, "Unable to fetch PodList resource")
-
-		return ctrl.Result{}, err
+		return fmt.Errorf("unable to fetch PodList resource: %w", err)
 	}
 
 	observedNodesInternal := make([]string, 0, len(observedPods.Items))
 	// nolint:gocritic // the copies are necessary for further redpandacluster updates
 	for _, item := range observedPods.Items {
-		observedNodesInternal = append(observedNodesInternal, fmt.Sprintf("%s.%s", item.Name, headlessSvc.HeadlessServiceFQDN()))
+		observedNodesInternal = append(observedNodesInternal, fmt.Sprintf("%s.%s", item.Name, internalFQDN))
```

Comment:
NIT: I just realised that I introduce wrong loop. We could switch from coping each Pod object and access them via index.

```suggestion
	for i := range observedPods.Items {
		observedNodesInternal = append(observedNodesInternal, fmt.Sprintf("%s.%s", observedPods.Items[i].Name, internalFQDN))
```